core/src/unprocessed_packet_batches.rs

```diff
@@ -34,6 +33,52 @@ pub struct DeserializedPacketBatch {
     pub unprocessed_packets: HashMap<usize, DeserializedPacket>,
 }
 
+pub struct UnprocessedPacketBatches(VecDeque<DeserializedPacketBatch>);
```

Comment:
Can you add doc comments here (`///`) to describe what this struct is used for? These are what I always look to first when looking at new code, so any additional context/information is very appreciated.